test/test_sklearn.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from typing import Callable, Type
+
+import platform
+from itertools import product
+
+from pynisher import (
+    CpuTimeoutException,
+    MemoryLimitException,
+    PynisherException,
+    WallTimeoutException,
+    contexts,
+    limit,
+    supports,
+)
+
+import pytest
+
+
+def train_svr(n_samples: int, n_features: int) -> None:
+    """Trains an svr model"""
+    from sklearn.datasets import make_regression
+    from sklearn.svm import SVR
+
+    X, y = make_regression(n_samples=n_samples, n_features=n_features)
+    SVR().fit(X, y)
+
+
+def train_svc(n_samples: int, n_features: int) -> None:
+    """Trains an svm model"""
+    from sklearn.datasets import make_classification
+    from sklearn.svm import SVC
+
+    X, y = make_classification(n_samples=n_samples, n_features=n_features)
+    SVC().fit(X, y)
+
+
+constraints_with_exceptions = [
+    (CpuTimeoutException, {"cpu_time": (1, "s")}),
+    (MemoryLimitException, {"memory": (500, "kB")}),
```

Comment:
This limit is a bit low, right? Shouldn't it rather be like 100MB and we use a larger dataset that triggers such an error in the SVM?

Reply:
Yup, I can do that but thankfully it illuminated some other issues that limiting `memory` doesn't always result in a `MemoryError`

Reply:
Maybe we could have two checks then? 

I assume you are writing about the import error, rights? How can it even get to the point of importing something with only 500kB if Python already eats up 37MB?

Reply:
Yup, it can't, nice spot. I increased the memory limit.

In general, I'm not sure we want to start consuming `ImportError` or `OSError` and converting to `MemoryLimitException`. There could be a perfectly legitimate case where `OSError` was raised because of file reading or something, and not due to a lack of memory.

I have two proposed solutions and I don't really like either of them:
* Inspect the errors to determine if it looks like a memory issue. This is not platform independent, as `test_sklearn` with `(500, "kb")` raised a `ImportError` on Linux and a `OSError` on windows. I've also seen a segmentation fault due to below. We can pick up on some of these specifics through inspecting errors and process exitcodes but something like that `ImportError` is non obvious.
```
OpenBLAS blas_thread_init: pthread_create failed for thread 2 of 8: Resource temporarily unavailable
OpenBLAS blas_thread_init: RLIMIT_NPROC 126908 current, 126908 max
```


* Just have an argument and let it be controlled. Upon catching one of these errors, convert it to a `MemoryLimitException`. I like this approach more as it's configurable and offloads specifics to the consumer of `pynisher`. With respect to the `sigsegv`, I'd rather not catch and handle this, just give a PynisherError as something non-graceful happened.
```python
# Defaults?
limit(memory_exceptions = [MemoryError, OSError, ImportError])
```